src/Api/ProjectsApi.php

```diff
@@ -75,4 +129,58 @@ public function projectsUserUserIdGet(string $userId, string $maxVersion = null,
   {
     // TODO: Implement projectsUserUserIdGet() method.
   }
+
+  /**
+   * @param Program[] $programs
+   *
+   * @return Project[]
+   */
+  private function getResponseData($programs)
+  {
+    $projects = [];
+    foreach ($programs as &$program)
+    {
+      $result = [
+        'id' => $program->getId(),
+        'name' => $program->getName(),
+        'author' => $program->getUser()->getUserName(),
+        'description' => $program->getDescription(),
+        'version' => $program->getCatrobatVersionName(),
+        'views' => $program->getViews(),
+        'download' => $program->getDownloads(),
+        'private' => $program->getPrivate(),
+        'flavor' => $program->getFlavor(),
+        'uploaded' => $program->getUploadedAt()->getTimestamp(),
+        'uploadedString' => $this->time_formatter->getElapsedTime($program->getUploadedAt()
+          ->getTimestamp()),
+        'screenshotLarge' => $this->program_manager->getScreenshotLarge($program->getId()),
+        'screenshotSmall' => $this->program_manager->getScreenshotSmall($program->getId()),
+        'projectUrl' => ltrim($this->generateUrl('program', [
+          'flavor' => $this->session->get('flavor_context'),
+          'id' => $program->getId(),
+        ]), '/'),
+        'downloadUrl' => ltrim($this->generateUrl('download', [
+          'id' => $program->getId(),
+        ]), '/'),
+        'filesize' => $program->getFilesize() / 1048576,
+      ];
+      $project = new Project($result);
+      array_push($projects, $project);
+    }
+
+    return $projects;
+  }
+
+  /**
+   * @param string $token
+   *
+   * @return mixed
+   */
+  private function decodeToken($token)
```

Comment:
This sounds to me like something more API calls will need in the future.
It would be nice to have this either in the user_manager available everywhere in the project. Or provide via a new service class.